### PR TITLE
Search and facet block for directory

### PR DESCRIPTION
**Pull request by Adnan-cds**
Provides a Directory search block.  It reuses the search block from the extra field.

### CODE REVIEW

**Comment by Adnan-cds** [comment]
Ready for review again.  To avoid the issue described in #34, do ```drush pm:install --yes localgov_directories_page``` instead of ```drush pm:install --yes localgov_directories``` while installing the localgov_directories module in a fresh localgov site.

Please note that as a result of the [fix](https://github.com/localgovdrupal/localgov_directories/pull/31/files#diff-06e5d87c1edbcfa82de0f0a23d5a6645) for #28, Directory channels now list the full Directory entry content instead of the teasers.  This is not related to this pull request.

@stephen-cox @andybroomfield @finnlewis

File: src/Plugin/Block/ChannelSearchBlock.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+   * @var \Drupal\Core\Entity\EntityRepositoryInterface
+   */
+  protected $entityRepository;
+
+  /**
+   * Form builder service.
+   *
+   * @var \Drupal\Core\Form\FormBuilderInterface
+   */
+  protected $formBuilder;
```
Comment (stephen-cox) [inline src/Plugin/Block/ChannelSearchBlock.php:230]: Very minor issue, but most classes are ordered as:
1. Class static variables
2. Class variables
4. Static methods
5. Constructor
6. Class methods

Not too bothered if you disagree :wink:

**Review by stephen-cox — APPROVED** [review]
The search form is wider than the sidebar so need to add a `width: 100%` to the `input` element. This could be done with something like:

```css
.block-localgov-directories-channel-search-block input[type=text] {
  width: 100%;
}
```

A couple of comments in the code, but otherwise looks fine.

File: css/search-block.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * @file
+ * Style rules for directory search.
+ */
+
+/* Hide the Directory search exposed form when rendered outside our block. */
+.view-localgov-directory-channel.view-display-id-node_embed > .view-filters,
+:not(.block-localgov-directories-channel-search-block) > .views-exposed-form.localgov-search-channel-primary,
+:not(.block-localgov-directories-channel-search-block) > .views-exposed-form.localgov-search-channel-secondary {
+  display: none;
+}
```
[the remark this answers is not in the document]
Comment (Adnan-cds) [reply css/search-block.css:11]: The "page" display of View allows splitting off the exposed filters into a separate block.  Perhaps you are thinking of that.  In this case we are using an *Embed* display of a View which made things slightly complicated :(

**Comment by Adnan-cds** [comment]
> The search form is wider than the sidebar so need to add a width: 100% to the input element.

Done :)

FAO @stephen-cox

File: src/Plugin/Block/ChannelSearchBlock.php (hunk 1 above)
Comment (Adnan-cds) [reply src/Plugin/Block/ChannelSearchBlock.php:230]: Hi Steve,
You are right that most classes follow this format.  In the absence of an agreed up on convention, I place the public methods first and the class variables last.  IIRC I picked up this convention from Pete Goodliffe's [Code craft](https://www.goodreads.com/book/show/44875.Code_Craft).  For the ChannelSearchBlock plugin class, the most important public method is ChannelSearchBlock::build().  Hence it's at the top.  The idea is, if you have to work on that block, you are most likely interested in its build() method.  So you better find it as early as possible.  I am happy to change the order.  Just let me know.

FAO @stephen-cox

Comment (stephen-cox) [reply src/Plugin/Block/ChannelSearchBlock.php:230]: @Adnan-cds   Drupal core has all variables at the top followed by constructor, although static methods seem to be mixid in. Here's an example: https://git.drupalcode.org/project/drupal/-/blob/9.0.x/core/lib/Drupal/Core/Entity/ContentEntityBase.php

It's not a major thing and it sounds like this is holding other work up so happy to approve.

Comment (Adnan-cds) [reply src/Plugin/Block/ChannelSearchBlock.php:230]: Thanks Steve :)

**Pull request merged**